### PR TITLE
Issue #13293: Add supported Java Language Level to website

### LINKED ISSUE #13293: Add supported Java Language Level to website
Discussed at https://github.com/checkstyle/checkstyle/issues/13279, it would be a good idea to point out what language level the project currently supports:

> @nrmancuso , do you think it is reasonable to put somewhere on our website info what version of jdk syntax we support currently ?
we have only run time - https://checkstyle.org/#JRE_and_JDK

_Originally posted by @romani in https://github.com/checkstyle/checkstyle/issues/13279#issuecomment-1602646621_

### PR DESCRIPTION
**Pull request by mahfouz72**
closes #13293

### CODE REVIEW

**Comment by mahfouz72** [comment]
github, generate site

File: src/xdocs/index.xml.vm [hunk 1]
```diff
@@ -217,6 +217,27 @@
       </div>
     </section>
 
+    <section name="Supported Language Level">
```
Comment (romani) [inline src/xdocs/index.xml.vm:220]: Why Level?

It is first time I see this term for syntax version.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs/index.xml.vm (hunk 1 above)
Comment (mahfouz72) [reply src/xdocs/index.xml.vm:220]: I am familiar with this term from the IDEA project structure settings. see the message you received [here](https://github.com/checkstyle/checkstyle/pull/15480#issuecomment-2281952480). Also I took it from issue title and description

Comment (mahfouz72) [reply src/xdocs/index.xml.vm:220]: `Supported Language features` ?

File: src/xdocs/index.xml.vm [hunk 2]
```diff
@@ -217,6 +217,27 @@
...
+      <p>
+        Checkstyle's grammar and checks currently support all Java language features introduced
+        in Java 21 and below (both regular and preview), except for:
```
Comment (nrmancuso) [inline src/xdocs/index.xml.vm:223]: I would prefer to not mention specific preview features at all, and just say that we may support preview features depending on community demand and that any preview feature support may change at any time.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
First pass:

**Comment by nrmancuso** [comment]
@mahfouz72 how can we remember to update this?

File: src/xdocs/index.xml.vm (hunk 1 above)
Comment (mahfouz72) [reply src/xdocs/index.xml.vm]: done

File: src/xdocs/index.xml.vm (hunk 2 above)
Comment (mahfouz72) [reply src/xdocs/index.xml.vm]: done

**Comment by mahfouz72** [comment]
> how can we remember to update this?

Maybe we add a reminder at the end of the `New Language Feature Check Integration Process` doc that after completing support (the check updates or creating a new check), update the "Supported Java Language Version" section in the relevant documentation?

**Comment by mahfouz72** [comment]
github, generate site

File: src/xdocs/index.xml.vm [hunk 3]
```diff
@@ -217,6 +217,19 @@
       </div>
     </section>
 
+    <section name="Supported Java Language Version">
+      <p>
+        Checkstyle's grammar and checks currently support all Java language features introduced
```
Comment (nrmancuso) [inline src/xdocs/index.xml.vm:222]: Let's make this only about what syntax we can parse, Check support will likely never be comprehensive.

File: src/xdocs/index.xml.vm [hunk 4]
```diff
@@ -217,6 +217,19 @@
...
+        in <a href="https://docs.oracle.com/javase/specs/jls/se21/html/index.html">Java 21</a>
+        and below. We may support preview features depending on community demand,
+        but please note that support for any preview feature may change at any time.
```
Comment (nrmancuso) [inline src/xdocs/index.xml.vm:225]:
```suggestion
        but please note that support for a given preview feature may change at any time.
```

Avoids redundancy here.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Items:

File: src/xdocs/index.xml.vm (hunk 4 above)
Comment (mahfouz72) [reply src/xdocs/index.xml.vm]: done

File: src/xdocs/index.xml.vm (hunk 3 above)
Comment (mahfouz72) [reply src/xdocs/index.xml.vm]: done

**Comment by mahfouz72** [comment]
github, generate site

**Review by nrmancuso — APPROVED** [review]

File: src/xdocs/index.xml.vm [hunk 5]
```diff
@@ -217,6 +217,19 @@
       </div>
     </section>
 
+    <section name="Supported Java Language Version">
+      <p>
+        Checkstyle can parse all Java language features introduced
+        in <a href="https://docs.oracle.com/javase/specs/jls/se21/html/index.html">Java 21</a>
+        and below. We may support preview features depending on community demand,
```
Comment (rnveach) [inline src/xdocs/index.xml.vm:224]: What about 22? Why did we stop at 21?

From our doc, unnamed patterns/variables were in 22.

Comment (mahfouz72) [reply src/xdocs/index.xml.vm:224]: done to 22.

**Comment by mahfouz72** [comment]
github, generate site

**Review by rnveach — APPROVED** [review]

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**